src/EditorFeatures/Core/Tagging/AsynchronousTaggerProvider.cs

```diff
@@ -3,79 +3,189 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Text;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Options;
 using Microsoft.CodeAnalysis.Editor.Shared.Tagging;
+using Microsoft.CodeAnalysis.Options;
 using Microsoft.CodeAnalysis.Shared.TestHooks;
 using Microsoft.VisualStudio.Text;
 using Microsoft.VisualStudio.Text.Editor;
 using Microsoft.VisualStudio.Text.Tagging;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Editor.Tagging
 {
     /// <summary>
-    /// Computes <see cref="ITag"/>s for a text buffer in an asynchronous manner.  Computation of
-    /// the tags is handled by a provided <see cref="IAsynchronousTaggerDataSource{TTag}"/>.  The
-    /// <see cref="AsynchronousTaggerProvider{TTag}"/> handles the jobs of scheduling when to 
-    /// compute tags, managing the collection of tags, and notifying the and keeping the user 
-    /// interface up to date with the latest tags produced.
+    /// Base type of all asynchronous tagger providers (<see cref="ITaggerProvider"/> and <see cref="IViewTaggerProvider"/>). 
     /// </summary>
-    internal class AsynchronousTaggerProvider<TTag> : ITaggerProvider
+    internal abstract partial class AsynchronousTaggerProvider<TTag> : IAsynchronousTaggerDataSource<TTag>, ITaggerProvider, IViewTaggerProvider
```

Comment:
This might lead people down a bad path where they think they have to implement both or export both or something. A comment of "export one or the other, but not both" might be wise. (or have derived types w/ the implementations of each? I can imagine you can do that cheaply in code.)